metricbeat/module/kubernetes/events/events.go

```diff
@@ -0,0 +1,129 @@
+package events
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+
+	"github.com/ericchiang/k8s"
+	"github.com/ghodss/yaml"
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("kubernetes", "events", New); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the MetricSet
+// The events MetricSet listens to events from Kubernetes API server and streams them to the output.
+// MetricSet implements the mb.PushMetricSet interface, and therefore does not rely on polling.
+type MetricSet struct {
+	mb.BaseMetricSet
+	watcher *EventWatcher
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	logp.Warn("EXPERIMENTAL: The kubernetes events metricset is experimental")
```

Comment:
Done. We would need to update the metricset generator as this was generated code.